File: cpp/sprint0/e.cpp
#include <iostream>
#include <vector>

using namespace std;

// Если ответ существует, верните вектор из двух элементов
// Если нет - то верните пустой вектор
vector <int> TwoSum(const vector <int> &sortedArr, int targetSum) {
	// Здесь реализация вашего решения

}

void print(const vector <int> &result) {
	if (result.size() != 2)
		cout << "None" << endl;
	else
		cout << result[0] << " " << result[1] << endl;
}

int main() {
	int n, targetSum;
	cin >> n;
	vector <int> sortedArr(n);
	for (int i = 0; i < n; ++i)
		cin >> sortedArr[i];
	cin >> targetSum;
	print(TwoSum(sortedArr, targetSum));
}

File: csharp/sprint0/E.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

public class E
{
    private static TextReader reader;
    private static TextWriter writer;

    // Если ответ существует, верните список из двух элементов
    // Если нет - то верните пустой список
    public static List<int> TwoSum(List<int> sortedArray, int targetSum)
    {
        // Здесь реализация вашего решения
    }

    public static void Main(string[] args)
    {
        reader = new StreamReader(Console.OpenStandardInput());
        writer = new StreamWriter(Console.OpenStandardOutput());

        var n = ReadInt();
        var sortedArray = ReadList();
        var targetSum = ReadInt();
        var result = TwoSum(sortedArray, targetSum);
        writer.WriteLine(result.Any() ? "None" : string.Join(" ", result));

        reader.Close();
        writer.Close();
    }

    private static int ReadInt()
    {
        return int.Parse(reader.ReadLine());
    }

    private static List<int> ReadList()
    {
        return reader.ReadLine()
            .Split(new[] { ' ', '\t', }, StringSplitOptions.RemoveEmptyEntries)
            .Select(int.Parse)
            .ToList();
    }
}
